src/Development/IDE/Spans/LocalBindings.hs

```diff
@@ -0,0 +1,73 @@
+{-# LANGUAGE DerivingStrategies         #-}
+
+module Development.IDE.Spans.LocalBindings
+  ( Bindings
+  , getLocalScope
+  , getFuzzyScope
+  , bindings
+  ) where
+
+import           Control.DeepSeq
+import           Data.IntervalMap.FingerTree (IntervalMap, Interval (..))
+import qualified Data.IntervalMap.FingerTree as IM
+import qualified Data.Map as M
+import qualified Data.Set as S
+import           Development.IDE.GHC.Compat (RefMap, identType, identInfo, getScopeFromContext, Scope(..), Name, Type)
+import           Development.IDE.Types.Location
+import           Development.IDE.GHC.Error
+import           SrcLoc
+import           NameEnv
+
+------------------------------------------------------------------------------
+-- | Turn a 'RealSrcSpan' into an 'Interval'.
+realSrcSpanToInterval :: RealSrcSpan -> Interval Position
+realSrcSpanToInterval rss =
+  Interval
+    (realSrcLocToPosition $ realSrcSpanStart rss)
+    (realSrcLocToPosition $ realSrcSpanEnd   rss)
+
+------------------------------------------------------------------------------
+-- | Compute which identifiers are in scope at every point in the AST. Use
+-- 'getLocalScope' to find the results.
+bindings :: RefMap -> Bindings
+bindings refmap = Bindings $ foldr (uncurry IM.insert) mempty  $ do
```

Comment:
This needs to go through the full list of name envs before it returns anything, right? If so, maybe it should be a `foldl'`